src/librustc/ty/context.rs

```diff
@@ -423,6 +423,10 @@ pub struct GlobalCtxt<'tcx> {
     pub hir: hir_map::Map<'tcx>,
     pub maps: maps::Maps<'tcx>,
 
+    pub shim_mir_cache: RefCell<FxHashMap<ty::InstanceDef<'tcx>,
+                                          &'tcx RefCell<Mir<'tcx>>>>,
+    pub mir_shimmer: RefCell<Option<Box<ty::MirShimmer<'tcx>>>>,
+
```

Comment:
Can you put a `FIXME` on these to eventually integrate them better with the rest of on-demand?